myadmin-angular/projects/temp-study-material-app/src/app/components/view-course-material/view-course-material.component.ts

```diff
@@ -31,7 +32,7 @@ export class ViewCourseMaterialComponent implements OnInit {
   }
 
   ngOnInit() {
-    this.sub = this.route.params.subscribe(params => {
+    this.routeParamsSubscription = this.route.params.subscribe(params => {
       this.academicYear = +params['academicYear']; // (+) converts string 'id' to a number
```

Comment:
I haven't seen any examples of that. Angular docs also uses a +
https://angular.io/guide/router#route-parameters-in-the-activatedroute-service

The + also works in plain js to convert a string to a number

Reply:
Yep I know and I don't have a problem with you doing it like that and you can leave it like that.

The difference is you are referencing plain old JavaScript but I am talking about TypeScript:
https://www.typescriptlang.org/docs/handbook/basic-types.html#type-assertions

And since we are essentially coding in TypeScript I was just wondering if type casting will work in this case.